modules/appeals_api/spec/services/appeals_api/uploads/document_validation_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe AppealsApi::Uploads::DocumentValidation do
+  describe '#validate' do
+    # rubocop:disable Layout/LineLength
+    let(:valid_pdf) { { 'document': Rack::Test::UploadedFile.new("#{::Rails.root}/modules/appeals_api/spec/fixtures/expected_10182_extra.pdf") } }
+    let(:non_pdf) { { 'document': Rack::Test::UploadedFile.new("#{::Rails.root}/modules/appeals_api/spec/fixtures/test_incorrect_type.docx") } }
+    let(:pretend_pdf) { { 'document': Rack::Test::UploadedFile.new("#{::Rails.root}/modules/appeals_api/spec/fixtures/test_fake_pdf.pdf") } }
+    let(:oversize_pdf) { { 'document': Rack::Test::UploadedFile.new("#{::Rails.root}/modules/appeals_api/spec/fixtures/oversize_11x17.pdf") } }
+
+    context 'when supported file uploaded (100MB 11in x 11in PDF)' do
+      it 'does not raise an error' do
+        expect { described_class.new(valid_pdf).validate }.not_to raise_error
+      end
+    end
+
+    context 'when unsupported file type uploaded' do
+      it 'raises an error when non-pdf file type' do
+        expect { described_class.new(non_pdf).validate }.to raise_error AppealsApi::Uploads::DocumentValidation::UploadValidationError
```

Comment:
Going to add checks for the actual text content of the errors found in the locale files.